File: nodejs/250113_B4_30585.js
/**
 * https://www.acmicpc.net/problem/30585
 */

const [ n, ...lines ] = require('fs').readFileSync(0, 'utf-8').toString().trim().split('\n');
const [ r, c ] = n.split(' ').map(Number);
let sum = [ ...lines.join('') ].map(ch => +ch).reduce((a, c) => a + c, 0);
console.log(Math.min(sum, r * c - sum));

File: python/250113_B4_30585.py
# https://www.acmicpc.net/problem/30585

[ r, c ] = map(int, input().split())
num = "".join([ input() for _ in range(r) ])
print( min( len(num.replace("0", "")), len(num) - len(num.replace("0", "")) ))
